clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -941,6 +941,39 @@ convertStringAttrToDenseElementsAttr(mlir::cir::ConstArrayAttr attr,
       llvm::ArrayRef(values));
 }
 
+template <typename StorageTy> StorageTy getZeroInitFromType(mlir::Type Ty);
+
+template <> mlir::APInt getZeroInitFromType(mlir::Type Ty) {
+  assert(Ty.isa<mlir::cir::IntType>() && "expected int type");
+  auto IntTy = Ty.cast<mlir::cir::IntType>();
+  return mlir::APInt::getZero(IntTy.getWidth());
+}
+
+template <> mlir::APFloat getZeroInitFromType(mlir::Type Ty) {
+  if (Ty.isF32())
+    return mlir::APFloat(0.f);
+  else if (Ty.isF64())
+    return mlir::APFloat(0.0);
+  else
+    llvm_unreachable("only float and double supported");
```

Comment:
I would suggest to use the same style for the both functions - either `llvm_unreachable` or `assert`.